docs/tutorials/python/sagas/index.md

```diff
@@ -0,0 +1,551 @@
+
+A `BookVacationInput` object is created with the extracted data, which will be passed to the Workflow.
+
+The Temporal client is obtained using the `get_temporal_client()` function.
+
+The Workflow is executed using `client.execute_workflow()`, passing the input object and other required parameters.
+Based on the result of the Workflow execution, a response is prepared and returned.
+If the booking process is cancelled, the response indicates this. Otherwise, it provides details about the booked car, hotel, and flight.
+
+**Start the Client**
+
+Now to start the Client, run the following command in your terminal:
+
+```bash
+python3 run_workflow.py
+```
+
+Once the Client is set up, you can start the booking process and see the Saga pattern in action.
+
+## Start the Booking Process
+
+To run the booking process, you can use the following `curl` command to send a `POST` request to the `/book` endpoint.
+This request will trigger the Workflow, and you will receive a response with the booking details or a cancellation message.
+
+```bash
+curl -X POST http://localhost:3002/book \
+-H "Content-Type: application/json" \
+-d '{
+    "name": "John Doe",
+    "attempts": 5,
+    "car": "valid-car-id",
+    "hotel": "valid-hotel-id",
+    "flight": "valid-flight-id"
+}'
+```
+
+**Output**
+
+The expected output should be a JSON response similar to the following:
+
+```json
+{
+  "cancelled": false,
+  "car": "Car: valid-car-id",
+  "flight": "Flight: valid-flight-id",
+  "hotel": "Hotel: valid-hotel-id",
+  "result": "Booked car: valid-car-id Booked hotel: valid-hotel-id Booked flight: valid-flight-id",
+  "user_id": "john-doe-184942"
+}
+```
+
+
+You've successfully initiated and completed your booking process using the Saga pattern with Temporal in Python.
+Next, you'll learn how to simulate errors to test the robustness of your implementation.
+
+### Simulate an error
+
+To ensure your implementation can handle failures gracefully, you will simulate a booking failure.
+This step will demonstrate how the Saga pattern with Temporal manages to rollback in case of errors.
+
+To simulate a booking failure, you can use the following `curl` command. T
+his request includes an invalid hotel booking ID, which will cause the booking process to fail and trigger the rollback process.
+
+```bash
+curl -X POST http://localhost:3002/book \
+-H "Content-Type: application/json" \
+-d '{
+    "name": "Jane Smith",
+    "attempts": 3,
+    "car": "valid-car-id",
+    "hotel": "invalid-hotel-id",
+    "flight": "valid-flight-id"
+}'
+```
+
+The value `invalid` will trigger an exception, causing the booking to rollback.
+
+**Output**
+
+The expected output should be a JSON response similar to the following:
+
+```json
+{
+  "cancelled": true,
+  "result": "Voyage cancelled",
+  "user_id": "jane-smith-609592"
+}
+```
+
+In this case, the booking process was cancelled due to the invalid hotel booking ID.
```

Comment:
How can we verify that these compensations ran? This looks like a place to open up the temporal web ui and show the event history to show they worked.